### PR TITLE
Suppress warnings in `tests/test_distributions.py`

### PR DESCRIPTION
**Pull request by nzw0301**
## Motivation

A part of https://github.com/optuna/optuna/issues/3815.

## Description of the changes

- Suppress expected warning messages in `test_distributions.py`.
- Fix docstring by following the coding style.

### CODE REVIEW

File: tests/test_distributions.py [hunk 1]
```diff
@@ -179,16 +179,20 @@ def test_check_distribution_compatibility() -> None:
...
     )
-    distributions.check_distribution_compatibility(
-        EXAMPLE_DISTRIBUTIONS["id"], distributions.IntDistribution(low=-3, high=2, step=2)
-    )
+    with warnings.catch_warnings():
+        # When `step` is 2, UserWarning will be raised since the range is not divisible by 2.
+        # The range will be replaced with [-3, 1].
+        warnings.simplefilter("ignore", category=UserWarning)
+        distributions.check_distribution_compatibility(
+            EXAMPLE_DISTRIBUTIONS["id"], distributions.IntDistribution(low=-3, high=2, step=2)
```
Comment (not522) [inline tests/test_distributions.py:194]: How about using `low=-3` and `high=1` to suppress `UserWarning`?

Comment (nzw0301) [reply tests/test_distributions.py]: Thanks! Definitely, it looks simpler.

File: tests/test_distributions.py [hunk 2]
```diff
@@ -121,7 +121,7 @@ def test_distribution_to_json() -> None:
 
 def test_check_distribution_compatibility() -> None:
 
-    # test the same distribution
+    # Test the same distribution.
```
Comment (toshihikoyanase) [inline tests/test_distributions.py:124]: Good catch. Thanks!

**Review by toshihikoyanase — APPROVED** [review]
LGTM! Thank you.

@not522 Could you continue the review, please?

**Review by not522 — APPROVED** [review]
LGTM!

**Pull request merged**